### DISCUSSION TITLE
WebView2  CoreWebView2 is null

### DISCUSSION THREAD

**Discussion by GHext** [MicrosoftEdge/WebView2Feedback#3979] [Q&A]
(1) webView2.EnsureCoreWebView2Async();
does not return anything  and  CoreWebView2 is already null.

(3) webView21.CoreWebView2InitializationCompleted()   event
the control's Microsoft.Web.WebView2.WinForms.WebView2.CoreWebView2  InitializationCompletedevent will be invoked before the task completes or on exceptions.

 await, WebView2  should return something.

**Comment by MysteriousGreenBean** [comment]
I had exact same issue in our Outlook VSTO Add-in, but it was cause by creatung webView2 environment on the background thread. When I switched it to the main thread, it started working as expected.

**Comment by LiangTheDev** [comment]
@GHext , is the above comment helpful? If not, could you share the details of the InitializationException of CoreWebView2InitializationCompletedEventArgs reported by CoreWebView2InitializationCompleted event?

**Comment by LiangTheDev** [comment]
What exception are you seeing? As WebView2 creation timeout takes more than 2 minutes, if we wait for 3 minutes, would there be any exceptions thrown? Do you see message printed out if you run the app under debugger?

Another common issue if that we are not setting [user data folder](https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/user-data-folder?tabs=dotnet) for WebView2 and the default user data folder is not writable by current user. You might want to try change await webView2.EnsureCoreWebView2Async(); to the following and see whether that makes any difference.

var env = await CoreWebView2Environment.CreateAsync(null, "C:\\temp");
await webView2.EnsureCoreWebView2Async(env);

**Comment by LiangTheDev** [comment]
Could you add WebView to UI (`CustomTask = Globals.ThisAddIn.CustomTaskPanes.Add(webView2, "ControlSetOption");`) before EnsureCoreWebView2Async? WebView2 requires an hwnd to be initialized which means it has to be in UI tree before initialization can be completed.

Also it should be `CoreWebView2Environment.CreateAsync(null, "C:\\temp")`, not "C:\temp" for the testing code. Note that using this temporary folder (c:\temp) is just for debugging. For real code, you should provide a proper folder specific to your app.

Another basic thing to ensure is that you do have Microsoft Edge WebView2 Runtime installed on the machine.,
